### PR TITLE
Adding tips and tricks

### PR DESCRIPTION
**Pull request by jesper-friis**

### CODE REVIEW

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: doc/tips_and_tricks.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    pip install -r requirements_dev.txt
+    pip install -r requirements_doc.txt
+
+    VIRTUAL_ENV=/path/to/virtual/env
+    source $VIRTUAL_ENV/bin/activate
+    mkdir build
+    cd build
+    cmake -DCMAKE_INSTALL_PREFIX=$VIRTUAL_ENV ..
+    make
+    make install
```
Comment (CasperWA) [inline doc/tips_and_tricks.md:21]:
```suggestion
```shell
pip install virtualenvwrapper
mkvirtualenv dlite
pip install -U pip
pip install -r requirements.txt
pip install -r requirements_dev.txt
pip install -r requirements_doc.txt

# VIRTUAL_ENV=/path/to/virtual/env
source $VIRTUAL_ENV/bin/activate
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=$VIRTUAL_ENV ..
make
make install
```
```

Comment (CasperWA) [reply doc/tips_and_tricks.md:21]: VIRTUAL_ENV skulle gerne allerede være automatisk sat når man er i sit virtuelle miljø. Når du installerer `virtualenvwrapper` kan du i stedet for `VIRTUAL_ENV` og `source` linjerne bruge: `workon dlite`.
Og i stedet for `mkvirtualenv` kan du bruge `mkproject`.

File: doc/tips_and_tricks.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+
```
Comment (CasperWA) [inline doc/tips_and_tricks.md:24]:
```suggestion
```

**Review by CasperWA — CHANGES_REQUESTED** [review]
Make code sections fenced sections with the given language for better lexing and presentation in the documentation (similar to my suggested change).

You have a lot of white space in this file as well. It's good practice to surround headers by a single white space line, and otherwise only have single white space lines throughout the document, not double.
Also it's good practice to use a single header syntax, instead of mixing two.

**Comment by jesper-friis** [comment]
> Make code sections fenced sections with the given language for better lexing and presentation in the documentation (similar to my suggested change).

There are no real benefits of syntax highlighting for the shell, so I prefer to keep the formatting as lean and readable as possible.

**Comment by jesper-friis** [comment]
> You have a lot of white space in this file as well. It's good practice to surround headers by a single white space line, and otherwise only have single white space lines throughout the document, not double.

We should use the flexibility of markdown to make the text as natural and easy to read in raw text as possible. Two blank lines before a section header makes the sectioning easy to distinguish from paragraphs. 

> Also it's good practice to use a single header syntax, instead of mixing two.
I don't agree. The double and single underline are much more natural ways to mark main headers. I agree that the triple hash is not the most natural way to mark a third-level header, but it is not so bad either, since it is quite distinct (especially if preceded with two blank lines).

File: doc/tips_and_tricks.md (hunk 1 above)
Comment (jesper-friis) [reply doc/tips_and_tricks.md]: > VIRTUAL_ENV skulle gerne allerede være automatisk sat når man er i sit virtuelle miljø. Når du installerer `virtualenvwrapper` kan du i stedet for `VIRTUAL_ENV` og `source` linjerne bruge: `workon dlite`. Og i stedet for `mkvirtualenv` kan du bruge `mkproject`.

Ja, det har du ret i. Har opdateret instruktionen. 

Hvad er fordelen med at bruke `mkproject`? Vi vil jo ikke lave et nyt projekt, kun et nyt virtuelt miljø.

Comment (CasperWA) [reply doc/tips_and_tricks.md]: Fordelen er mere automatisering. Det er det samme, bare at den laver en ny mappe udenfor alle de "rå" virtualenv filer/mapper. En "arbejdsmappe". Det er bare dét. Og så CD'er den også automatisk ind i den mappe.

File: doc/tips_and_tricks.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+To run the image `cp37-manylinux_x86_64` do
+
+    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest CIBW_BUILD=cp37-manylinux_x86* python -m cibuildwheel --output-dir wheelhouse --platform linux python
+
+which should run the tests and hopefully fail at the same place as on
+GitHub.  If that is the case, you can run the image again, but pause
+it before running the tests by adding prepending
+`CIBW_BEFORE_TEST=cat` to the previous command:
+
+    CIBW_BEFORE_TEST=cat CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest CIBW_BUILD=cp37-manylinux_x86* python -m cibuildwheel --output-dir wheelhouse --platform linux python
```
Comment (CasperWA) [inline doc/tips_and_tricks.md:59]:
```suggestion
    CIBW_BEFORE_TEST=cat \
    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest \
    CIBW_BUILD=cp37-manylinux_x86* \
    python -m cibuildwheel \
    --output-dir wheelhouse \
    --platform linux \
    python
```

File: doc/tips_and_tricks.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest CIBW_BUILD=cp37-manylinux_* python -m cibuildwheel --print-build-identifiers --platform linux python
+
+This should write
+
+    cp37-manylinux_x86_64
+    cp37-manylinux_i686
+
+To run the image `cp37-manylinux_x86_64` do
+
+    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest CIBW_BUILD=cp37-manylinux_x86* python -m cibuildwheel --output-dir wheelhouse --platform linux python
```
Comment (CasperWA) [inline doc/tips_and_tricks.md:52]:
```suggestion
    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest \
    CIBW_BUILD=cp37-manylinux_x86* \
    python -m cibuildwheel \
    --output-dir wheelhouse \
    --platform linux \
    python
```

File: doc/tips_and_tricks.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+### Debugging within the docker environment
+First, enter your virtual environment with dlite and install cibuildwheel.
+
+    workon dlite
+    pip install cibuildwheel
+
+To list all manylinux images for Python 3.7, do
+
+    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest CIBW_BUILD=cp37-manylinux_* python -m cibuildwheel --print-build-identifiers --platform linux python
```
Comment (CasperWA) [inline doc/tips_and_tricks.md:43]:
```suggestion
    CIBW_MANYLINUX_X86_64_IMAGE=ghcr.io/sintef/dlite-python-manylinux2014_x86_64:latest \
    CIBW_BUILD=cp37-manylinux_* \
    python -m cibuildwheel \
    --print-build-identifiers \
    --platform linux \
    python
```

**Review by CasperWA — CHANGES_REQUESTED** [review]

**Review by CasperWA — APPROVED** [review]
It would be good to write up your intended formatting, syntax, and linting requirements for documentation for documentation contributors.
This can be done by writing a "contribution" page in the documentation.

**Pull request merged**